text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,189 @@
+when versions of dependencies are added or removed. The table below assumes that upstream dependencies
+follow [semantic versioning](https://semver.org/). For those that do not, maintainers may determine
+their versioning independently.
+
+| Dependency change | Buildpack Version expectation |
+|-|-|
+| ADD Patch | Bump Patch |
+| ADD Minor | - Bump Minor if set as default<br>  version<br>- Bump Patch otherwise |
+| ADD Major | - Bump Major if set as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Patch | Bump Patch |
+| REMOVE Minor | - Bump Major if unset as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Major | Bump Major |
+<br/>
+
+
+#### Composite Buildpack
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Implementation buildpack major bumps | Implementation buildpack minor bumps | Implementation buildpack patch bumps |
+| Removal of an order group dropping<br>support of some workflow | Addition of an order group that do<br>not cause change in detection logic |  |
+| Swapping order of order groups |  |  |
+| Addition of an order group<br>causing change in detection logic |  |  |
+<br/>
+
+
+#### Builder
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Lifecycle major bumps | Minor bumps of components | Patch bumps of components |
+| Stack Image major bumps | Addition of a buildpack<br>as last in the order |  |
+| Buildpack major bumps |  |  |
+| Change in Stack ID |  |  |
+| Removal of a Buildpack |  |  |
+| Swapping order of buildpacks |  |  |
+| Addition of a buildpack except<br>as last in the order |  |  |
+<br/>
+
+#### Stack
+| `Major`                 | `Minor`                   | `Patch`                   |
+|-------------------------|---------------------------|---------------------------|
+| Removal of  a package   | Addition of a new package | Update of a package       |
+
+### Intial Development
+
+A version of `0.0.x` is considered to be in initial development and may not be
+suitable for user consumption.
+
+## User Policy
+
+* **Patch** bump: User should update their buildpack/builder without
+	hesitation.
+
+* **Minor** bump: User can update their buildpack/builder to avail new
+	features. Nothing will break if buildpack-set default dependency versions are
+	not overridden.
+
+* **Major** bump: Users must review changelog and test build system on update.
+	May contain breaking changes.
+
+
+### Critical security updates
+
+If a user is on the latest `Major` and `Minor` versions, critical security
+updates when they occur should be available for users on the latest `Patch`
+version.
+
+## Process
+
+This section proposes a process that will help maintainers easily determine
+whether a given change or a set of changes require a bump in the major, minor,
+or patch component.
+
+* Changes should be accepted to the main line of code only via Pull Requests.
+
+* All Pull Requests (including automated PRs by the
+  [paketo-bot](https://github.com/paketo-bot)) must be tagged with either of
```

Comment:
Also not sure if this gets too in the weeds, but I want to call out that implementing this RFC will require new automation that figures out how to correctly label bot PRs.

Some different cases that will need to be implemented:
- a dependency update in an implementation buildpack's `buildpack.toml` (e.g. `go-dist` loses go 1.15, gains go 1.17)

- an implementation buildpack update in a composite `buildpack.toml` (e.g. paketo-buildpacks/go-build v3.1 replaces v3.0 in the paketo-buildpacks/go buildpack)

- lifecycle version is updated in a builder
- build image version is updated in a builder
- buildpack version is updated in a builder